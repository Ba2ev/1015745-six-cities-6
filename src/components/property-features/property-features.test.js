import React from 'react';
import {render} from '@testing-library/react';
import PropertyFeatures from './property-features';

const offer = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13
    }
  },
  "previewImage": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`
  ],
  "title": `The house among olive `,
  "isFavorite": true,
  "isPremium": true,
  "rating": 3,
  "type": `room`,
  "bedrooms": 1,
  "maxAdults": 1,
  "price": 169,
  "goods": [
    `Breakfast`,
    `Air conditioning`,
    `Laptop friendly workspace`,
    `Washer`
  ],
  "host": {
    "id": 25,
    "name": `Angelina`,
    "isPro": true,
    "avatarUrl": `img/avatar-angelina.jpg`
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16
  },
  "id": 1
};

it(`PropertyFeatures should render correctly`, () => {
  const {getByText} = render(<PropertyFeatures type={offer.type}
    bedrooms={offer.bedrooms}
    maxAdults={offer.maxAdults} />);

  const textType = getByText(offer.type);
  const textBedrooms = getByText(/Bedrooms/i);
  const textMaxAdults = getByText(/adults/i);

  expect(textType).toBeInTheDocument();
  expect(textBedrooms).toBeInTheDocument();
  expect(textMaxAdults).toBeInTheDocument();
});
