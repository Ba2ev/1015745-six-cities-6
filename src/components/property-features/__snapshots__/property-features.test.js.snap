// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyFeatures should render snapshot correctly 1`] = `
<div>
  <ul
    class="property__features"
  >
    <li
      class="property__feature property__feature--entire"
    >
      room
    </li>
    <li
      class="property__feature property__feature--bedrooms"
    >
      1 Bedrooms
    </li>
    <li
      class="property__feature property__feature--adults"
    >
      Max 1 adults
    </li>
  </ul>
</div>
`;
